eNMS/services/configuration/netconf.py

```diff
@@ -0,0 +1,304 @@
+from json import dumps
+from sqlalchemy import Boolean, ForeignKey, Integer
+import xmltodict
+from wtforms.widgets import TextArea
+
+from eNMS.database import db
+from eNMS.forms.automation import ConnectionForm
+from eNMS.forms.fields import (
+    BooleanField,
+    HiddenField,
+    IntegerField,
+    SelectField,
+    StringField,
+)
+from eNMS.models.automation import ConnectionService
+
+
+class NetconfService(ConnectionService):
+    __tablename__ = "netconf_service"
+    pretty_name = "NETCONF"
+    id = db.Column(
+        Integer,
+        ForeignKey("connection_service.id"),
+        primary_key=True,
+    )
+    nc_type = db.Column(db.SmallString)
+    target = db.Column(db.SmallString)
+    default_operation = db.Column(db.SmallString)
+    test_option = db.Column(db.SmallString)
+    error_option = db.Column(db.SmallString)
+    xml_filter = db.Column(db.LargeString, default="")
+    timeout = db.Column(Integer, default=15)
+    lock = db.Column(Boolean, default=False)
+    unlock = db.Column(Boolean, default=False)
+    commit_conf = db.Column(Boolean, default=False)
+    copy_source = db.Column(db.SmallString)
+    source_url = db.Column(db.SmallString)
+    copy_destination = db.Column(db.SmallString)
+    destination_url = db.Column(db.SmallString)
+    xml_conversion = db.Column(Boolean, default=True)
+
+    __mapper_args__ = {"polymorphic_identity": "netconf_service"}
+
+    def job(self, run, payload, device=None):
+        xml_filter = run.sub(run.xml_filter, locals())
+        run.log(
+            "info",
+            "Sending NETCONF request",
+            device,
+            logger="security",
+        )
+        result = {"success": False, "result": "No NETCONF operation selected."}
```

Comment:
is this useful ? Is it possible to not select any netconf operation from the dropdown list ?